File: chap05/run-model5-5.R
library(cmdstanr)

d <- read.csv(file='input/data-shopping-3.csv')
d$Income <- d$Income/100
data <- list(V=nrow(d), Sex=d$Sex, Income=d$Income, Dis=d$Discount, Y=d$Y)

model <- cmdstan_model(stan_file='model/model5-5.stan')
# model <- cmdstan_model(stan_file='model/model5-5b.stan')
fit <- model$sample(data=data, seed=123, parallel_chains=4)
fit$save_object(file='output/result-model5-5.RDS')

File: chap05/run-model5-5.py
import pandas
import cmdstanpy

d = pandas.read_csv('input/data-shopping-3.csv')
d.Income /= 100
d.rename(columns={'Discount': 'Dis'}, inplace=True)
data = d.to_dict('list')
data.update({'V':len(d)})

model = cmdstanpy.CmdStanModel(stan_file='model/model5-5.stan')
# model = cmdstanpy.CmdStanModel(stan_file='model/model5-5b.stan')
fit = model.sample(data=data, seed=123, parallel_chains=4)
fit.save_csvfiles('output/result-model5-5')
